plugins/processors/noise/noise_test.go

```diff
@@ -0,0 +1,165 @@
+package noise
+
+import (
+	"math"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/testutil"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/rand"
+	"gonum.org/v1/gonum/stat/distuv"
+)
+
+// Helper function which returns a map for metrics for easy lookup
+func createTestMetrics() map[string]telegraf.Metric {
+	m := make(map[string]telegraf.Metric)
+	m["cpu"] = testutil.MustMetric(
+		"cpu",
+		map[string]string{},
+		map[string]interface{}{
+			"usage_guest":  2.5,
+			"usage_system": 1.5,
+			"usage_nice":   0.5,
+			"usage_irq":    1.0,
+		},
+		time.Unix(0, 0),
+	)
+	m["disk"] = testutil.MustMetric(
+		"disk",
+		map[string]string{},
+		map[string]interface{}{
+			"free":        250,
+			"inodes_free": 1500.0,
+			"inodes_used": 1337,
+		},
+		time.Unix(0, 0),
+	)
+	return m
+}
+
+// Verifies that field values are modified by the Laplace noise
+func TestAddNoiseToMetric(t *testing.T) {
+	generators := []string{"laplace", "gaussian", "uniform"}
+	for _, generator := range generators {
+		p := Noise{
+			NoiseType: generator,
+			Scale:     1.0,
+			Mu:        0.0,
+			Sigma:     0.2,
+			Min:       -1,
+			Max:       1,
+			Log:       testutil.Logger{},
+		}
+		_ = p.Init()
+		metrics := testutil.MockMetrics()
+		after := p.Apply(metrics[0].Copy())[0]
+		require.NotEqual(t, metrics[0], after)
+	}
+}
+
+// Verifies that no int64 or uint64 overflow occurs while adding noise
+func TestAddNoiseOverflowCheck(t *testing.T) {
+	p := Noise{
+		NoiseType: "laplace",
+		Scale:     1.0,
+		Log:       testutil.Logger{},
+	}
+	_ = p.Init()
+	p.Generator = &distuv.Laplace{Mu: p.Mu, Scale: p.Scale, Src: rand.NewSource(4)}
+	require.Equal(t, math.MaxInt64, p.addNoise(math.MaxInt64))
+
+	// rand should be -1.3528540286500519, a underflow happens for uint64
+	require.Equal(t, uint64(0), p.addNoise(uint64(math.MaxUint64)))
+
+	// rand should be 0.813394603644003, a positive overflow happens
+	p.Generator = &distuv.Laplace{Mu: p.Mu, Scale: p.Scale, Src: rand.NewSource(2)}
+	require.Equal(t, math.MaxInt64, p.addNoise(math.MaxInt64))
+	require.Equal(t, uint64(math.MaxUint64), p.addNoise(uint64(math.MaxUint64)))
+}
```

Comment:
How about using table-based tests here? This eases addition of new tests...
```suggestion
type testDistribution {
	value float64
}

func (t *testDistribution) Rand() float64 {
	return t.value
}

func TestAddNoiseOverflowCheck(t *testing.T) {
	tests := []struct {
		name         string
		input        []telegraf.Metric
		expected     []telegraf.Metric
		distribution distuv.Rander

	}{
		name: "underflow",
		input: []telegraf.Metric{
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": math.MinInt64},
				time.Unix(0, 0),
			),
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": uint64(0)},
				time.Unix(0, 0),
			),
		},
		expected: []telegraf.Metric{
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": math.MinInt64},
				time.Unix(0, 0),
			),
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": uint64(0)},
				time.Unix(0, 0),
			),
		},
		distribution: &testDistribution{value: -1.0},
	},
	{
		name: "overflow",
		input: []telegraf.Metric{
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": math.MaxInt64},
				time.Unix(0, 0),
			),
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": math.MaxUint64},
				time.Unix(0, 0),
			),
		},
		expected: []telegraf.Metric{
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": math.MaxInt64},
				time.Unix(0, 0),
			),
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{"value": math.MaxUint64},
				time.Unix(0, 0),
			),
		},
		distribution: &testDistribution{value: 1.0},
	},	
	{
		name: "non-numeric fields",
		input: []telegraf.Metric{
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{
					"a": "test",
					"b": true,
				},
				time.Unix(0, 0),
			),
		},
		expected: []telegraf.Metric{
			testutil.MustMetric("cpu",
				map[string]string{},
				map[string]interface{}{
					"a": "test",
					"b": true,
				},
				time.Unix(0, 0),
			),
		},
		distribution: &testDistribution{value: 1.0},
	}	

	for _, tt := range tests {
		t.Run(tt.name, func(t *testing.T) {
			plugin := Noise{
				NoiseType: "laplace",
				Scale:     1.0,
				Log:       testutil.Logger{},
			}
			require.NoError(t, plugin.Init())
			plugin.generator = tt.distribution
			actual := plugin.Apply(tt.inputs)
			testutil.RequireMetricsEqual(t, tt.expected, actual)
		})
	}
}
```

Reply:
Looks way better, thanks!